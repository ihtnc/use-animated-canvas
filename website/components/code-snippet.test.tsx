import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import CodeSnippet from './code-snippet'
import typescript from 'highlight.js/lib/languages/typescript'

describe('CodeSnippet component', () => {
  test('should render collapsible code with typescript styles by default', () => {
    const code = 'const message = "Hello, world!";'
    const { container } = render(<CodeSnippet
      code={code}
      language='typescript'
      languageFn={typescript}
    />)

    expect(container).toMatchSnapshot()
  })

  test('should render collapsible code with typescript styles', () => {
    const code = 'const message = "Hello, world!";'
    const { container } = render(<CodeSnippet
      code={code}
      language='typescript'
      languageFn={typescript}
      collapsible={true}
    />)

    expect(container).toMatchSnapshot()
  })

  test('should render code with typescript styles', () => {
    const code = 'const message = "Hello, world!";'
    const { container } = render(<CodeSnippet
      code={code}
      language='typescript'
      languageFn={typescript}
      collapsible={false}
    />)

    expect(container).toMatchSnapshot()
  })

  test('should apply hljs-specific styles', () => {
    const code = 'const message = "Hello, world!";'
    const { container } = render(<CodeSnippet
      code={code}
      language='typescript'
      languageFn={typescript}
    />)

    const keywordValue = container.querySelector('.hljs-keyword')
    const stringValue = container.querySelector('.hljs-string')
    expect(keywordValue).not.toBeNull()
    expect(stringValue).not.toBeNull()
  })

  test('should render collapsible sections when specified', () => {
    const code = 'const message = "Hello, world!";'
    const { queryByText } = render(<CodeSnippet
      code={code}
      language='typescript'
      languageFn={typescript}
      collapsible={true}
    />)

    const collapseButton = queryByText('Code')
    expect(collapseButton).not.toBeNull()
  })

  test('should not render collapsible sections when specified', () => {
    const code = 'const message = "Hello, world!";'
    const { queryByText } = render(<CodeSnippet
      code={code}
      language='typescript'
      languageFn={typescript}
      collapsible={false}
    />)

    const collapseButton = queryByText('Code')
    expect(collapseButton).toBeNull()
  })
})