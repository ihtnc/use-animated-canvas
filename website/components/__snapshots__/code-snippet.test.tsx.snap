// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeSnippet component > should render code with typescript styles 1`] = `
<div>
  <div
    class="rounded-lg w-full max-h-[75dvh] md:max-h-fit p-2 bg-gradient-to-b from-gray-300 dark:from-neutral-700 to-transparent overflow-auto"
  >
    <pre>
      <code>
        <span
          class="hljs-keyword"
        >
          const
        </span>
         message = 
        <span
          class="hljs-string"
        >
          "Hello, world!"
        </span>
        ;
      </code>
    </pre>
  </div>
</div>
`;

exports[`CodeSnippet component > should render collapsible code with typescript styles 1`] = `
<div>
  <div
    class="pb-4 w-full"
    data-headlessui-state="open"
    data-open=""
  >
    <h3
      aria-controls="headlessui-disclosure-panel-:r3:"
      aria-expanded="true"
      class="flex w-fit group text-xl font-semibold cursor-pointer"
      data-headlessui-state="open"
      data-open=""
      id="headlessui-disclosure-button-:r2:"
    >
      Code 
      <span
        class="inline-block transition-transform motion-reduce:transform-none
          group-data-[open]:-rotate-90
          group-data-[open]:pl-1
          group-data-[open]:pt-1
          group-hover:group-data-[open]:-translate-y-1
          group-hover:translate-y-1
          rotate-90
          pb-1"
      >
        -&gt;
      </span>
    </h3>
    <div
      data-headlessui-state="open"
      data-open=""
      id="headlessui-disclosure-panel-:r3:"
    >
      <div
        class="rounded-lg w-full max-h-[75dvh] md:max-h-fit p-2 bg-gradient-to-b from-gray-300 dark:from-neutral-700 to-transparent overflow-auto"
      >
        <pre>
          <code>
            <span
              class="hljs-keyword"
            >
              const
            </span>
             message = 
            <span
              class="hljs-string"
            >
              "Hello, world!"
            </span>
            ;
          </code>
        </pre>
      </div>
    </div>
  </div>
</div>
`;

exports[`CodeSnippet component > should render collapsible code with typescript styles by default 1`] = `
<div>
  <div
    class="pb-4 w-full"
    data-headlessui-state="open"
    data-open=""
  >
    <h3
      aria-controls="headlessui-disclosure-panel-:r1:"
      aria-expanded="true"
      class="flex w-fit group text-xl font-semibold cursor-pointer"
      data-headlessui-state="open"
      data-open=""
      id="headlessui-disclosure-button-:r0:"
    >
      Code 
      <span
        class="inline-block transition-transform motion-reduce:transform-none
          group-data-[open]:-rotate-90
          group-data-[open]:pl-1
          group-data-[open]:pt-1
          group-hover:group-data-[open]:-translate-y-1
          group-hover:translate-y-1
          rotate-90
          pb-1"
      >
        -&gt;
      </span>
    </h3>
    <div
      data-headlessui-state="open"
      data-open=""
      id="headlessui-disclosure-panel-:r1:"
    >
      <div
        class="rounded-lg w-full max-h-[75dvh] md:max-h-fit p-2 bg-gradient-to-b from-gray-300 dark:from-neutral-700 to-transparent overflow-auto"
      >
        <pre>
          <code>
            <span
              class="hljs-keyword"
            >
              const
            </span>
             message = 
            <span
              class="hljs-string"
            >
              "Hello, world!"
            </span>
            ;
          </code>
        </pre>
      </div>
    </div>
  </div>
</div>
`;
